File: TellDontAsk/AlarmsObjectOriented/Criterias/OutsideWeekendCriteria.cs
using System;
using AlarmsObjectOriented.Interfaces;

namespace AlarmsObjectOriented.Criterias
{
  public class OutsideWeekendCriteria : TimeCriteria
  {
    public bool IsSatisfied()
    {
      return !new WeekendCriteria().IsSatisfied();
    }

    public void Output()
    {
      Console.WriteLine("it's not weekend");
    }
  }
}

File: Java/AlarmsObjectOriented/src/main/java/criterias/OutsideWeekendCriteria.java
package criterias;

import interfaces.TimeCriteria;

public class OutsideWeekendCriteria implements TimeCriteria {
    public boolean isSatisfied() {
      return !new WeekendCriteria().isSatisfied();
    }

    public void output() {
      System.out.println("it's not weekend");
    }
  }
